android/src/main/kotlin/com/adyen/checkout/flutter/components/instant/InstantComponentManager.kt

```diff
@@ -132,7 +188,58 @@ class InstantComponentManager(
         )
     }
 
-    private fun handleAction(action: Action) = instantPaymentComponent?.handleAction(action, activity)
+    // We delete the ideal component integration when the instant component supports ideal.
+    private fun createIdealAdvancedComponent(
+        configuration: CheckoutConfiguration,
+        paymentMethod: PaymentMethod,
+        componentId: String,
+    ): IdealComponent {
+        return IdealComponent.PROVIDER.get(
+            activity = activity,
+            paymentMethod = paymentMethod,
+            checkoutConfiguration = configuration,
+            callback =
+                IdealComponentAdvancedCallback(
+                    componentFlutterInterface,
+                    componentId,
+                    ::hideLoadingBottomSheet
+                ),
+            key = UUID.randomUUID().toString()
+        )
+    }
+
+    private fun createIdealSessionComponent(
+        configuration: CheckoutConfiguration,
+        paymentMethod: PaymentMethod,
+        componentId: String,
+    ): IdealComponent {
+        val sessionSetupResponse = SessionSetupResponse.SERIALIZER.deserialize(sessionHolder.sessionSetupResponse)
+        val order = sessionHolder.orderResponse?.let { Order.SERIALIZER.deserialize(it) }
+        val checkoutSession =
+            CheckoutSession(
+                sessionSetupResponse = sessionSetupResponse,
+                order = order,
+                environment = configuration.environment,
+                clientKey = configuration.clientKey
+            )
+        return IdealComponent.PROVIDER.get(
+            activity = activity,
+            checkoutSession = checkoutSession,
+            paymentMethod = paymentMethod,
+            checkoutConfiguration = configuration,
+            componentCallback =
+                IdealComponentSessionCallback(
+                    componentFlutterInterface,
+                    componentId,
+                    ::handleAction,
+                    ::hideLoadingBottomSheet
+                ),
+            key = UUID.randomUUID().toString()
```

Comment:
Just checking: Are you intentionally attempting to create a new instance every time?

Reply:
Yes, this is intentionally to support multiple simulations components. Would you suggest to use only one instance?

Reply:
Not necessarily. Unless merchants would want to access to the same Ideal Component which is not possible at this point. At this point if this aligns with iOS behavior, it's perfectly fine. I would only recommend to consider if in the future flutter sdk would support this or not.

Reply:
Alright, I would suggest to keep it unique for now. Whenever we might get the demand to reuse or access the instant component, we could offer it. But for now I think its fine.